test/plugins/sails.spec.js

```diff
@@ -0,0 +1,90 @@
+'use strict'
+
+const agent = require('./agent')
+const plugin = require('../../src/plugins/sails')
+
+wrapIt()
+
+describe('Plugin', () => {
+  let tracer
+  let Sails
+
+  describe('sails', () => {
+    withVersions(plugin, 'sails', version => {
+      beforeEach(() => {
+        tracer = require('../..')
+      })
+
+      before(() => {
+        return agent.load(plugin, 'sails').then(() => {
+          Sails = require(`../../versions/sails@${version}`).get()
+          Sails.lift()
+        })
+      })
+
+      after(() => {
+        Sails.lower()
+
+        agent.close()
+      })
+
+      it('traces registered actions', done => {
+        Sails.registerAction(() => { return 'test response' }, 'testAction')
+
+        expect(Sails.getActions()['testaction']()).to.eq('test response')
```

Comment:
though it's really unlikely to flake, I think the convention of adding the expect handlers for trace content before initiating the traced action is done for a reason.